src/v/cluster/tx_gateway.cc

```diff
@@ -97,7 +97,12 @@ ss::future<begin_group_tx_reply> tx_gateway::begin_group_tx(
 
 ss::future<prepare_group_tx_reply> tx_gateway::prepare_group_tx(
   prepare_group_tx_request&& request, rpc::streaming_context&) {
-    return _rm_group_proxy->prepare_group_tx_locally(std::move(request));
+    return _rm_group_proxy->prepare_group_tx(
```

Comment:
imho it's better to throw an exception from here rather than shifting this responsibility to _rm_group_proxy

Reply:
yes, done.